src/avram/database.cr

```diff
@@ -39,6 +39,22 @@ abstract class Avram::Database
     new.delete
   end
 
+  # Listens for `pg_notify()` calls on each channel in `channels`
+  # Yields a `PQ::Notification` object with `channel`, `payload`, and `pid`.
+  #
+  # ```
+  # # pg_notify("callback", "123")
+  # AppDatabase.listen("callback", "jobs") do |notification|
+  #   notification.channel # => "callback"
+  #   notification.payload # => "123"
+  # end
+  # ```
+  def self.listen(*channels : String, &block : PQ::Notification ->) : Nil
+    new.listen(*channels) do |notification|
+      block.call(notification)
+    end
```

Comment:
```suggestion
    new.listen(*channels, &block)
```

I think that works?

Reply:
It does, but I actually did this way intentionally thinking that maybe it works better for documentational purposes when people are looking at the code. I wanted to use `yield`, but Crystal blows up because the block is actually a proc 😒  This was me just trying to be clever 😅  but I'll revert it.

Reply:
@jwoertink just because you have `&block` as a parameter you can still call yield.

So it could be:

```crystal
new.listen(*channels) do |notification|
  yield notification
end
```

Same goes for all the other places if that's what you want

Reply:
I tried that, but I get this error:

```
In src/avram/database.cr:54:7

 54 | yield notification
      ^
Error: can't use `yield` inside a proc literal or captured block

Make sure to read the whole docs section about blocks and procs,
including "Capturing blocks" and "Block forwarding":

https://crystal-lang.org/reference/syntax_and_semantics/blocks_and_procs.html
```

It's all good though.